### PR TITLE
#1582: added the endpoint and started doing the tests

### LINKED ISSUE #1582: Finance Fixes - Edit Reimbursement Endpoint
### Description

Currently once a user submits a refund, they can't edit it. This means if they make a type they can never correct it. We wanna add the ability for a user to edit a refund after they have submitted it, to ensure that all the refunds saved in Finishline are accurate. This ticket is to add the backend endpoint to facilitate that feature.

### Acceptance Criteria

- [x] This endpoint should be on the route '/reimburse/:reimbursementId/edit'
- [x] This endpoint should take a date and a int (refund amount in cents) as a payload
- [x] Only the person who made the refund should be able to edit it
- [x] Unit tests for this endpoint should be added to the backend

### Proposed Solution
Look at the edit expense type endpoint for an example

### PR DESCRIPTION
**Pull request by martin0he**
## Changes

I added an endpoint to change a reimbursement that only allows the creator to edit their reimbursement. It takes in an amount and date as a payload.

## Notes

Currently adding backend tests

## Test Cases
- when the user isn't the one who submitted the refund
- when the reimbursement doesn't exist
- when all the conditions have been met

## Screenshots
Case: when the user is not the submitter of the reimbursement:
image
Success Case: when the user is the submitter and the reimbursement exists
image
Case: when the reimbursement doesn't exist
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1582

### CODE REVIEW

File: src/backend/tests/reimbursement-requests.test.ts [hunk 1]
```diff
@@ -827,6 +827,60 @@ describe('Reimbursement Requests', () => {
...
+        ReimbursementRequestService.editReimbursement(
+          'fakeId',
+          reimbursementMock.userSubmitted,
+          reimbursementMock.amount,
+          reimbursementMock.dateCreated
+        )
+      ).rejects.toThrow(new NotFoundException('Reimbursement', 'fakeId'));
+    });
+
+    test('Successfully edits a reimbursement', async () => {
```
Comment (walker-sean) [inline src/backend/tests/reimbursement-requests.test.ts:864]: You need to add a mock for prisma.reimbursement.update to this test

**Review by walker-sean — CHANGES_REQUESTED** [review]

File: src/backend/tests/reimbursement-requests.test.ts [hunk 2]
```diff
@@ -827,6 +827,65 @@ describe('Reimbursement Requests', () => {
     });
   });
 
+  describe('Edit Reimbursement', () => {
+    const reimbursementMock = {
```
[the remark this answers is not in the document]
Comment (martin0he) [reply src/backend/tests/reimbursement-requests.test.ts:831]: all the other test data comes from reimbursement-requests.test-data.ts , should I still add it to just the seed or that file?

[the remark this answers is not in the document]
Comment (RChandler234) [reply src/backend/tests/reimbursement-requests.test.ts:831]: just put it in reimbursement-requests.test-data.ts

File: src/backend/tests/test-data/reimbursement-requests.test-data.ts [hunk 3]
```diff
@@ -232,3 +232,17 @@ export const sharedGiveMeMyMoney: ReimbursementRequest = {
...
+export const reimbursementMock = {
+  reimbursementId: 'reimbursementMockId',
+  purchaserId: batman.userId,
+  amount: 12,
+  dateCreated: new Date('2023-01-01'),
+  userSubmitted: batman,
+  userSubmittedId: batman.userId
+};
+
+export const updatedReimbursementMock = {
```
Comment (walker-sean) [inline src/backend/tests/test-data/reimbursement-requests.test-data.ts:245]: This is only used in one place, so you don't even need to declare this constant - you can just use the spread operator in the test - that way it is clear what youre doing

**Review by walker-sean — CHANGES_REQUESTED** [review]
ok last thing from me i think !!!

**Review by walker-sean — APPROVED** [review]
Nice!

**Pull request merged**